### DISCUSSION TITLE
Trigger usage

### DISCUSSION THREAD

**Discussion by danies8** [quartznet/quartznet#2402] [Q&A]
Hi,
Does the start date time and end date time oveeride the cron expression?
Why they are  separate ?
trigger = TriggerBuilder.Create().WithIdentity(jobName, groupName).StartAt(startDateTime)
        .WithSchedule(cronExpression).EndAt(endDateTime).Build();
Thanks,

**Comment by lahma** [comment]
You can experiment and check with `TriggerUtils` what will the fire times be.

**Reply by danies8** [reply]
I used https://www.freeformatter.com/cron-expression-generator-quartz.html
to evaluate cron expression. Does start time && end time override the cron expression.
I.E in cron expression i need to put * for month and * for year and will determine the start month/year and end month/year.

**Reply by lahma** [reply]
> You can experiment and check with TriggerUtils what will the fire times be.

**Reply by danies8** [reply]
@lahma 
I tried TriggerUtils.ComputeFireTimes(what i need to put here?)
this is my trigger:
trigger = TriggerBuilder.Create().WithIdentity(jobName, groupName).StartAt(startDateTime)
.WithSchedule(cronExpression).EndAt(endDateTime).Build();

**Reply by lahma** [reply]
Maybe you can Google this. This is getting quite time-consuming considering this is basically.. the basics.
